compiler/frontend/src/org/jetbrains/kotlin/resolve/calls/inference/ConstraintSystemImpl.kt

```diff
@@ -175,8 +188,8 @@ internal class ConstraintSystemImpl(
 
     override fun toBuilder(filterConstraintPosition: (ConstraintPosition) -> Boolean): ConstraintSystem.Builder {
         val result = ConstraintSystemBuilderImpl()
-        for ((typeParameter, typeBounds) in allTypeParameterBounds) {
-            result.allTypeParameterBounds.put(typeParameter, typeBounds.filter(filterConstraintPosition))
+        for (typeParameter in allTypeParameterBounds.entries) {
```

Comment:
Since `entries` returns a destructurable class, I suppose you can keep the `for ((typeParameter, typeBounds) ...` syntax here